src/main/java/org/testeditor/fixture/web/json/BrowserSettingsManager.java

```diff
@@ -40,36 +43,153 @@
      * </code>
      */
     public static List<BrowserSetupElement> getBrowserSettings() {
-        
         BrowserSetupReader reader = new BrowserSetupReader();
         List<BrowserSetupElement> elements = reader.readElements(FILE_NAME);
-
         Platform currentPlattform = Platform.getCurrent();
-        return getBrowserSpecificSetting(elements, currentPlattform);
-
+        return getBrowserElementsWithoutDuplicateEntries(elements,currentPlattform);
     }
     
+    /**
+     * This method is for testing purposes.
+     * @param fileName 
+     *          of JSON test file for browserSetup.json.
+     * @return 
+     *          A List of BrowserSetupElement
+     */
+    protected static List<BrowserSetupElement> getBrowserSettings(String fileName) {
+        BrowserSetupReader reader = new BrowserSetupReader();
+        List<BrowserSetupElement> elements = reader.readElements(fileName);
+        Platform currentPlattform = Platform.getCurrent();
+        return getBrowserElementsWithoutDuplicateEntries(elements,currentPlattform);
+    }
     
-    private static List<BrowserSetupElement> getBrowserSpecificSetting(List<BrowserSetupElement> 
-        elements,Platform platform) {
-        List<BrowserSetupElement> settings = new ArrayList<>();    
-        settings = separateBrowserSettingsForOs(elements, platform);
-        separateBrowserSettingsOsUnspecific(settings, elements);
-        return settings;
+    protected static List<BrowserSetupElement> getBrowserElementsWithoutDuplicateEntries(
+            List<BrowserSetupElement> elements,Platform platform) {
+        List<BrowserSetupElement> separatedBrowserSettingsForOs = separateBrowserSettingsForOs(elements, platform);
```

Comment:
do you really need the prefix 'separated' here? Given the naming, it is very hard to see that the caps and options are simply combined. maybe you could name the variables a bit more intention revealing. swapping the parameters in copOptionsIntoMap may help, too.